### PR TITLE
Feature: get node position

### PR DESCRIPTION
**Pull request by Wybxc**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -372,6 +372,19 @@ impl<T> Snarl<T> {
         }
     }
 
+    /// Returns reference to the node and its position.
+    pub fn get_node_pos(&self, idx: NodeId) -> Option<(&T, &Pos2)> {
+        self.nodes.get(idx.0).map(|node| (&node.value, &node.pos))
```
Comment (zakarumych) [inline src/lib.rs:377]: Please, write function in the same style

Comment (Wybxc) [reply src/lib.rs:377]: Clippy warns about "Manual implementation of `Option::map`" in this function

Comment (zakarumych) [reply src/lib.rs:377]: Odd, it doesn't do so in the "mut" version

Comment (zakarumych) [reply src/lib.rs:377]: Either way, please, keep style consistent.
If you worry about `clippy::manual_map` - fix it everywhere.

**Comment by pepperoni505** [comment]
Hi, any update on this PR?

**Comment by zakarumych** [comment]
Required methods are added here 081f7f3ba809e7eee43517386094167e50097fb3

**Pull request closed without merging**